src/CompoundRegion.cc

```diff
@@ -151,21 +159,43 @@ Circle UnionRegion::getBoundingCircle() const {
 }
 
 bool UnionRegion::contains(UnitVector3d const &v) const {
-    return getOperand(0).contains(v) || getOperand(1).contains(v);
+    for (auto&& operand: operands()) {
+        if (operand->contains(v)) {
+            return true;
+        }
+    }
+    return false;
 }
 
 Relationship UnionRegion::relate(Region const &rhs) const {
-    auto r1 = getOperand(0).relate(rhs);
-    auto r2 = getOperand(1).relate(rhs);
-    return
-        // Both operands must be disjoint with the given region for the union
+    auto result = DISJOINT | WITHIN;
+    // When result becomes CONTAINS we can stop checking.
+    auto const stop = CONTAINS;
+    for (auto&& operand: operands()) {
+        auto rel = operand->relate(rhs);
+        // All operands must be disjoint with the given region for the union
         // to be disjoint with it.
-        ((r1 & DISJOINT) & (r2 & DISJOINT))
-        // Both operands must be within the given region for the union to be
+        if ((rel & DISJOINT) != DISJOINT) {
+            result &= ~DISJOINT;
+        }
+        // All operands must be within the given region for the union to be
         // within it.
-        | ((r1 & WITHIN) & (r2 & WITHIN))
-        // If either operand contains the given region, the union contains it.
-        | ((r1 & CONTAINS) | (r2 & CONTAINS));
+        if ((rel & WITHIN) != WITHIN) {
+            result &= ~WITHIN;
+        }
+        // If any operand contains the given region, the union contains it.
+        // NOTE: The logic here is incomplete there may be cases when union
+        // of regions can fully contain a region even though individual regions
+        // do not contain it.
```

Comment:
Thankfully this is still consistent with the carefully conservative `relate` documentation:
https://github.com/lsst/sphgeom/blob/781aacb6e4d47f4a80a54d607375a7f372c316b2/include/lsst/sphgeom/Region.h#L124-L127

Reply:
Indeed, I read `relate` documentation later, but forgot to remove my dummy comments.